From 279142812fc385691e82e2ef3308fce5de47246a Mon Sep 17 00:00:00 2001
From: Mohsen Dolaty <mohsen.dolaty@amd.com>
Date: Tue, 18 Jan 2022 22:53:43 -0600
Subject: [PATCH 1/1] linux-aspeed: Add I3C DIMM and CPU RAS devices to DTS

Add I3C DIMM (SPD) and CPU RAS (APML) devices to SP5 DTS

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts     | 245 ++++++---
 arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts   | 483 +++++++++++++-----
 arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts     | 222 +++++++-
 arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts | 388 ++++++++++++++
 4 files changed, 1127 insertions(+), 211 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index f300522ba7ff..6449c3c875ec 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -544,80 +544,201 @@ tmp468@48 {
 	};
 };

+#ifdef OPENBMC_I3C_ENABLE
+
 &i3c0 {
 	// P0 DIMM (A-F) SPD Access
 	status = "okay";

 	jdec-spd;

-	//DIMM A
-	spd5118_0_0: spd@50 {
-		compatible = "idt,spd";
-		reg = <0x50 0x0 0x00000000>;
-	};
-	//DIMM B
-	spd5118_0_1: spd@51 {
-		compatible = "idt,spd";
-		reg = <0x51 0x0 0x00000000>;
-	};
-	//DIMM C
-	spd5118_0_2: spd@52 {
-		compatible = "idt,spd";
-		reg = <0x52 0x0 0x00000000>;
-	};
-	//DIMM D
-	spd5118_0_3: spd@53 {
-		compatible = "idt,spd";
-		reg = <0x53 0x0 0x00000000>;
-	};
-	//DIMM E
-	spd5118_0_4: spd@54 {
-		compatible = "idt,spd";
-		reg = <0x54 0x0 0x00000000>;
-	};
-	//DIMM F
-	spd5118_0_5: spd@55 {
-		compatible = "idt,spd";
-		reg = <0x55 0x0 0x00000000>;
-	};
+        //DIMM A
+        spd5118_0_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_0_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_0_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_0_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_0_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_0_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_0_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_0_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_0_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_0_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_0_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_0_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_0_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_0_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_0_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_0_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_0_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_0_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
 };

 &i3c1 {
 	// P0 DIMM (G-L) SPD Access
 	status = "okay";

-	jdec_spd;
-	//DIMM G
-	spd5118_1_0: spd@50 {
-		compatible = "idt,spd";
-		reg = <0x50 0x0 0x00000000>;
-	};
-	//DIMM H
-	spd5118_1_1: spd@51 {
-		compatible = "idt,spd";
-		reg = <0x51 0x0 0x00000000>;
-	};
-	//DIMM I
-	spd5118_1_2: spd@52 {
-		compatible = "idt,spd";
-		reg = <0x52 0x0 0x00000000>;
-	};
-	//DIMM J
-	spd5118_1_3: spd@53 {
-		compatible = "idt,spd";
-		reg = <0x53 0x0 0x00000000>;
-	};
-	//DIMM K
-	spd5118_1_4: spd@54 {
-		compatible = "idt,spd";
-		reg = <0x54 0x0 0x00000000>;
-	};
-	//DIMM L
-	spd5118_1_5: spd@55 {
-		compatible = "idt,spd";
-		reg = <0x55 0x0 0x00000000>;
-	};
-};
+	jdec-spd;
+
+        //DIMM G
+        spd5118_1_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_1_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_1_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_1_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_1_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_1_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_1_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_1_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_1_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_1_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_1_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_1_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_1_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_1_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_1_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_1_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_1_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_1_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+};
+
+&i3c4 {
+    // P0 APML
+    status = "okay";
+
+    jdec-spd;
+
+    imx3112_p0: i3cmux@70,4CC00000000 {
+        reg = <0x70 0x4CC 0x00000000>;
+        assigned-address = <0x70>;
+    };
+    sbtsi_p0_0: sbtsi@4c,22400000000 {
+        reg = <0x4c 0x224 0x00000000>;
+        assigned-address = <0x4c>;
+    };
+    sbrmi_p0_1: sbrmi@3c,22400000001 {
+        reg = <0x3c 0x224 0x00000001>;
+        assigned-address = <0x3c>;
+    };
+};
+
+#endif  // OPENBMC_I3C_ENABLE

 &i2c2 {
 	// P0 APML
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
index fb1298882405..48b11801ad1a 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-quartz.dts
@@ -658,81 +658,180 @@ &i2c15 {
 	// RoT
 	status = "okay";
 };
+
 #ifdef OPENBMC_I3C_ENABLE
+
 &i3c0 {
 	// P0 DIMM (A-F) SPD Access
 	status = "okay";

 	jdec-spd;

-	//DIMM A
-	spd5118_0_0: spd@50 {
-		compatible = "idt,spd";
-		reg = <0x50 0x0 0x00000000>;
-	};
-	//DIMM B
-	spd5118_0_1: spd@51 {
-		compatible = "idt,spd";
-		reg = <0x51 0x0 0x00000000>;
-	};
-	//DIMM C
-	spd5118_0_2: spd@52 {
-		compatible = "idt,spd";
-		reg = <0x52 0x0 0x00000000>;
-	};
-	//DIMM D
-	spd5118_0_3: spd@53 {
-		compatible = "idt,spd";
-		reg = <0x53 0x0 0x00000000>;
-	};
-	//DIMM E
-	spd5118_0_4: spd@54 {
-		compatible = "idt,spd";
-		reg = <0x54 0x0 0x00000000>;
-	};
-	//DIMM F
-	spd5118_0_5: spd@55 {
-		compatible = "idt,spd";
-		reg = <0x55 0x0 0x00000000>;
-	};
+        //DIMM A
+        spd5118_0_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_0_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_0_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_0_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_0_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_0_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_0_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_0_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_0_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_0_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_0_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_0_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_0_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_0_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_0_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_0_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_0_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_0_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
 };

 &i3c1 {
 	// P0 DIMM (G-L) SPD Access
 	status = "okay";

-	jdec_spd;
-	//DIMM G
-	spd5118_1_0: spd@50 {
-		compatible = "idt,spd";
-		reg = <0x50 0x0 0x00000000>;
-	};
-	//DIMM H
-	spd5118_1_1: spd@51 {
-		compatible = "idt,spd";
-		reg = <0x51 0x0 0x00000000>;
-	};
-	//DIMM I
-	spd5118_1_2: spd@52 {
-		compatible = "idt,spd";
-		reg = <0x52 0x0 0x00000000>;
-	};
-	//DIMM J
-	spd5118_1_3: spd@53 {
-		compatible = "idt,spd";
-		reg = <0x53 0x0 0x00000000>;
-	};
-	//DIMM K
-	spd5118_1_4: spd@54 {
-		compatible = "idt,spd";
-		reg = <0x54 0x0 0x00000000>;
-	};
-	//DIMM L
-	spd5118_1_5: spd@55 {
-		compatible = "idt,spd";
-		reg = <0x55 0x0 0x00000000>;
-	};
-};
+	jdec-spd;
+
+        //DIMM G
+        spd5118_1_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_1_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_1_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_1_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_1_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_1_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_1_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_1_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_1_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_1_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_1_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_1_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_1_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_1_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_1_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_1_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_1_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_1_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+ };

 &i3c2 {
 	// P1 DIMM (A-F) SPD Access
@@ -740,74 +839,212 @@ &i3c2 {

 	jdec-spd;

-	//DIMM A
-	spd5118_2_0: spd@50 {
-		compatible = "idt,spd";
-		reg = <0x50 0x0 0x00000000>;
-	};
-	//DIMM B
-	spd5118_2_1: spd@51 {
-		compatible = "idt,spd";
-		reg = <0x51 0x0 0x00000000>;
-	};
-	//DIMM C
-	spd5118_2_2: spd@52 {
-		compatible = "idt,spd";
-		reg = <0x52 0x0 0x00000000>;
-	};
-	//DIMM D
-	spd5118_2_3: spd@53 {
-		compatible = "idt,spd";
-		reg = <0x53 0x0 0x00000000>;
-	};
-	//DIMM E
-	spd5118_2_4: spd@54 {
-		compatible = "idt,spd";
-		reg = <0x54 0x0 0x00000000>;
-	};
-	//DIMM F
-	spd5118_2_5: spd@55 {
-		compatible = "idt,spd";
-		reg = <0x55 0x0 0x00000000>;
-	};
+        //DIMM A
+        spd5118_2_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_2_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_2_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_2_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_2_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_2_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_2_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_2_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_2_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_2_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_2_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_2_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_2_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_2_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_2_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_2_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_2_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_2_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
 };

 &i3c3 {
 	// P1 DIMM (G-L) SPD Access
 	status = "okay";

-	jdec_spd;
-	//DIMM G
-	spd5118_3_0: spd@50 {
-		compatible = "idt,spd";
-		reg = <0x50 0x0 0x00000000>;
-	};
-	//DIMM H
-	spd5118_3_1: spd@51 {
-		compatible = "idt,spd";
-		reg = <0x51 0x0 0x00000000>;
-	};
-	//DIMM I
-	spd5118_3_2: spd@52 {
-		compatible = "idt,spd";
-		reg = <0x52 0x0 0x00000000>;
-	};
-	//DIMM J
-	spd5118_3_3: spd@53 {
-		compatible = "idt,spd";
-		reg = <0x53 0x0 0x00000000>;
-	};
-	//DIMM K
-	spd5118_3_4: spd@54 {
-		compatible = "idt,spd";
-		reg = <0x54 0x0 0x00000000>;
-	};
-	//DIMM L
-	spd5118_3_5: spd@55 {
-		compatible = "idt,spd";
-		reg = <0x55 0x0 0x00000000>;
-	};
+	jdec-spd;
+
+        //DIMM G
+        spd5118_3_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_3_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_3_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_3_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_3_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_3_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_3_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_3_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_3_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_3_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_3_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_3_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_3_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_3_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_3_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_3_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_3_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_3_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
 };
+
+&i3c4 {
+    // P0 APML
+    status = "okay";
+
+    jdec-spd;
+
+    imx3112_p0: i3cmux@70,4CC00000000 {
+        reg = <0x70 0x4CC 0x00000000>;
+        assigned-address = <0x70>;
+    };
+    sbtsi_p0_0: sbtsi@4c,22400000000 {
+        reg = <0x4c 0x224 0x00000000>;
+        assigned-address = <0x4c>;
+    };
+    sbrmi_p0_1: sbrmi@3c,22400000001 {
+        reg = <0x3c 0x224 0x00000001>;
+        assigned-address = <0x3c>;
+    };
+};
+
+&i3c5 {
+    // P1 APML
+    status = "okay";
+
+    jdec-spd;
+
+    imx3112_P1: i3cmux@70,4CC00000000 {
+        reg = <0x70 0x4CC 0x00000000>;
+        assigned-address = <0x70>;
+    };
+    sbtsi_p1_0: sbtsi@4c,22400000000 {
+        reg = <0x4c 0x224 0x00000000>;
+        assigned-address = <0x4c>;
+    };
+    sbrmi_p1_1: sbrmi@3c,22400000001 {
+        reg = <0x3c 0x224 0x00000001>;
+        assigned-address = <0x3c>;
+    };
+ };
+
 #endif  // OPENBMC_I3C_ENABLE

 &i2c2 {
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
index 9865ba754622..5d47c3dafd1c 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-ruby.dts
@@ -164,17 +164,6 @@ i2cswitch@70 {
 #endif //OPENBMC_LCD_DRIVER
 };

-&i2c2 {
-	// P0 APML - i2c10
-	status = "okay";
-	bus-frequency = <400000>;
-
-	sbtsi@4c {
-		compatible = "amd,sbtsi";
-		reg = <0x4c>;
-	};
-};
-
 // Misc Regulators
 &i2c4 {
 	// Net name i2c1
@@ -440,6 +429,202 @@ &i2c15 {
 	status = "okay";
 };

+#ifdef OPENBMC_I3C_ENABLE
+
+&i3c0 {
+        // P0 DIMM (A-F) SPD Access
+        status = "okay";
+
+        jdec-spd;
+
+        //DIMM A
+        spd5118_0_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_0_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_0_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_0_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_0_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_0_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_0_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_0_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_0_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_0_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_0_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_0_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_0_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_0_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_0_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_0_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_0_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_0_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+ };
+
+ &i3c1 {
+	// P0 DIMM (G-L) SPD Access
+	status = "okay";
+
+	jdec-spd;
+
+        //DIMM G
+        spd5118_1_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_1_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_1_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_1_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_1_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_1_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_1_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_1_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_1_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_1_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_1_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_1_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_1_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_1_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_1_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_1_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_1_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_1_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+ };
+
+&i3c4 {
+    // P0 APML
+    status = "okay";
+
+    jdec-spd;
+
+    imx3112_p0: i3cmux@70,4CC00000000 {
+        reg = <0x70 0x4CC 0x00000000>;
+        assigned-address = <0x70>;
+    };
+    sbtsi_p0_0: sbtsi@4c,22400000000 {
+        reg = <0x4c 0x224 0x00000000>;
+        assigned-address = <0x4c>;
+    };
+    sbrmi_p0_1: sbrmi@3c,22400000001 {
+        reg = <0x3c 0x224 0x00000001>;
+        assigned-address = <0x3c>;
+    };
+};
+
+#endif  // OPENBMC_I3C_ENABLE
+
 &i2c2 {
 	// P0 APML
 	status = "okay";
@@ -455,21 +640,6 @@ sbrmi@3c {
 	};
 };

-&i2c3 {
-	// P1 APML
-	status = "okay";
-
-	sbtsi@48 {
-		compatible = "amd,sbtsi";
-		reg = <0x48>;
-	};
-	// sbrmi
-	sbrmi@38 {
-		compatible = "amd,sbrmi";
-		reg = <0x38>;
-	};
-};
-
 &espi_ctrl {
 	status = "okay";

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
index 560631be5af2..98497fc246a8 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-titanite.dts
@@ -145,6 +145,394 @@ &i2c1 {
 		//<TBD> Define LCD details here
 };

+#ifdef OPENBMC_I3C_ENABLE
+
+&i3c0 {
+        // P0 DIMM (A-F) SPD Access
+        status = "okay";
+
+        jdec-spd;
+
+        //DIMM A
+        spd5118_0_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_0_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_0_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_0_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_0_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_0_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_0_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_0_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_0_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_0_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_0_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_0_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_0_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_0_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_0_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_0_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_0_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_0_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+ };
+
+ &i3c1 {
+	// P0 DIMM (G-L) SPD Access
+	status = "okay";
+
+	jdec-spd;
+
+        //DIMM G
+        spd5118_1_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_1_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_1_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_1_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_1_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_1_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_1_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_1_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_1_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_1_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_1_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_1_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_1_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_1_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_1_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_1_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_1_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_1_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+ };
+
+&i3c2 {
+        // P1 DIMM (A-F) SPD Access
+        status = "okay";
+
+	jdec-spd;
+
+        //DIMM A
+        spd5118_2_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_2_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_2_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM B
+        spd5118_2_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_2_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_2_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM C
+        spd5118_2_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_2_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_2_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM D
+        spd5118_2_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_2_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_2_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM E
+        spd5118_2_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_2_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_2_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM F
+        spd5118_2_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_2_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_2_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+ };
+
+&i3c3 {
+	// P1 DIMM (G-L) SPD Access
+	status = "okay";
+
+	jdec-spd;
+
+        //DIMM G
+        spd5118_3_0: spd@50,3C000000000 {
+                reg = <0x50 0x3C0 0x00000000>;
+                assigned-address = <0x50>;
+        };
+        pmic5xxx_3_0: pmic@48,20400000000 {
+                reg = <0x48 0x204 0x00000000>;
+                assigned-address = <0x48>;
+        };
+        rcd_3_0: rcd@58,2C000000000 {
+                reg = <0x58 0x2C0 0x00000000>;
+                assigned-address = <0x58>;
+        };
+        //DIMM H
+        spd5118_3_1: spd@51,3C000000001 {
+                reg = <0x51 0x3C0 0x00000001>;
+                assigned-address = <0x51>;
+        };
+        pmic5xxx_3_1: pmic@49,20400000001 {
+                reg = <0x49 0x204 0x00000001>;
+                assigned-address = <0x49>;
+        };
+        rcd_3_1: rcd@59,2C000000001 {
+                reg = <0x59 0x2C0 0x00000001>;
+                assigned-address = <0x59>;
+        };
+        //DIMM I
+        spd5118_3_2: spd@52,3C000000002 {
+                reg = <0x52 0x3C0 0x00000002>;
+                assigned-address = <0x52>;
+        };
+        pmic5xxx_3_2: pmic@4a,20400000002 {
+                reg = <0x4a 0x204 0x00000002>;
+                assigned-address = <0x4a>;
+        };
+        rcd_3_2: rcd@5a,2C000000002 {
+                reg = <0x5a 0x2C0 0x00000002>;
+                assigned-address = <0x5a>;
+        };
+        //DIMM J
+        spd5118_3_3: spd@53,3C000000003 {
+                reg = <0x53 0x3C0 0x00000003>;
+                assigned-address = <0x53>;
+        };
+        pmic5xxx_3_3: pmic@4b,20400000003 {
+                reg = <0x4b 0x204 0x00000003>;
+                assigned-address = <0x4b>;
+        };
+        rcd_3_3: rcd@5b,2C000000003 {
+                reg = <0x5b 0x2C0 0x00000003>;
+                assigned-address = <0x5b>;
+        };
+        //DIMM K
+        spd5118_3_4: spd@54,3C000000004 {
+                reg = <0x54 0x3C0 0x00000004>;
+                assigned-address = <0x54>;
+        };
+        pmic5xxx_3_4: pmic@4c,20400000004 {
+                reg = <0x4c 0x204 0x00000004>;
+                assigned-address = <0x4c>;
+        };
+        rcd_3_4: rcd@5c,2C000000004 {
+                reg = <0x5c 0x2C0 0x00000004>;
+                assigned-address = <0x5c>;
+        };
+        //DIMM L
+        spd5118_3_5: spd@55,3C000000005 {
+                reg = <0x55 0x3C0 0x00000005>;
+                assigned-address = <0x55>;
+        };
+        pmic5xxx_3_5: pmic@4d,20400000005 {
+                reg = <0x4d 0x204 0x00000005>;
+                assigned-address = <0x4d>;
+        };
+        rcd_3_5: rcd@5d,2C000000005 {
+                reg = <0x5d 0x2C0 0x00000005>;
+                assigned-address = <0x5d>;
+        };
+};
+
+&i3c4 {
+    // P0 APML
+    status = "okay";
+
+    jdec-spd;
+
+    imx3112_p0: i3cmux@70,4CC00000000 {
+        reg = <0x70 0x4CC 0x00000000>;
+        assigned-address = <0x70>;
+    };
+    sbtsi_p0_0: sbtsi@4c,22400000000 {
+        reg = <0x4c 0x224 0x00000000>;
+        assigned-address = <0x4c>;
+    };
+    sbrmi_p0_1: sbrmi@3c,22400000001 {
+        reg = <0x3c 0x224 0x00000001>;
+        assigned-address = <0x3c>;
+    };
+};
+
+&i3c5 {
+    // P1 APML
+    status = "okay";
+
+    jdec-spd;
+
+    imx3112_P1: i3cmux@70,4CC00000000 {
+        reg = <0x70 0x4CC 0x00000000>;
+        assigned-address = <0x70>;
+    };
+    sbtsi_p1_0: sbtsi@4c,22400000000 {
+        reg = <0x4c 0x224 0x00000000>;
+        assigned-address = <0x4c>;
+    };
+    sbrmi_p1_1: sbrmi@3c,22400000001 {
+        reg = <0x3c 0x224 0x00000001>;
+        assigned-address = <0x3c>;
+    };
+ };
+
+#endif  // OPENBMC_I3C_ENABLE
+
 &i2c2 {
 	// P0 APML - i2c10
 	status = "okay";
--
2.25.1
